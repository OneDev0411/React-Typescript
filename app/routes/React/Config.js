// Config.js
import React from 'react'
import { Route, IndexRoute, Redirect } from 'react-router'

// Main component
import App from '../../components/App'

// Pages
import Landing from '../../components/Pages/Landing'
import Recents from '../../components/Pages/Dashboard/Recents'
import Mls from '../../components/Pages/Dashboard/Mls'
import Contacts from '../../components/Pages/Dashboard/Contacts'
import Tasks from '../../components/Pages/Dashboard/Tasks'
import Transactions from '../../components/Pages/Dashboard/Transactions'
import SingleTransaction from '../../components/Pages/Dashboard/Transactions/Single'
import NewTransaction from '../../components/Pages/Dashboard/Transactions/New'
import SignUp from '../../components/Pages/SignUp'
import SignIn from '../../components/Pages/SignIn'
import Verify from '../../components/Pages/Verify'
import Password from '../../components/Pages/Password'
import NoMatch from '../../components/Pages/NoMatch'
import Settings from '../../components/Pages/Account/Settings'
import Notifications from '../../components/Pages/Account/Notifications'
import Recommend from '../../components/Pages/Dashboard/Mls/Listing/Recommend'
import Agents from '../../components/Pages/Dashboard/Mls/Agents'

export default (
  <Route path="/" component={App}>
    <IndexRoute component={Landing}/>
    <Route path="signup" component={SignUp}/>
    <Route path="signin" component={SignIn}/>
    <Route path="account/settings" component={Settings}/>
    <Route path="account/notifications" component={Notifications}/>
    <Redirect from="dashboard" to="dashboard/recents" />
    <Route path="dashboard/recents" component={Recents}/>
    <Route path="dashboard/recents/:room_id" component={Recents}/>
    <Route path="dashboard/mls" component={Mls}/>
    <Route path="dashboard/mls/agents" component={Agents}/>
    <Route path="dashboard/mls/listing/recommend" component={Recommend}/>
    <Route path="dashboard/contacts" component={Contacts}/>
    <Route path="dashboard/contacts/:id" component={Contacts}/>
    <Route path="dashboard/tasks" component={Tasks}/>
    <Route path="dashboard/transactions" component={Transactions}/>
    <Route path="dashboard/transactions/new" component={NewTransaction}/>
    <Route path="dashboard/transactions/:id" component={SingleTransaction}/>
    <Route path="dashboard/transactions/:id/attachments/:id" component={SingleTransaction}/>
    <Route path="verify/:slug" component={Verify}/>
    <Route path="password/:slug" component={Password}/>
    <Route path="*" component={NoMatch}/>
  </Route>
)