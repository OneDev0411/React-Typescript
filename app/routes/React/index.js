// index.js
import React from 'react'
import { match, RoutingContext } from 'react-router'
import ReactDOMServer from 'react-dom/server'

// Store
import AppStore from '../../stores/AppStore'

// Config
import routes from './Config'

module.exports = (app, config) => {

  app.get('*',(req, res) => {

    match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
      
      let reactMarkup = ReactDOMServer.renderToStaticMarkup(<RoutingContext data={AppStore.data} {...renderProps} />)
      
      res.locals.reactMarkup = reactMarkup

      console.log(reactMarkup)

      if (error) {
      
        res.status(500).send(error.message)
      
      } else if (redirectLocation) {
        
        res.redirect(302, redirectLocation.pathname + redirectLocation.search)

      } else if (renderProps) {
        
        // Success!
        res.status(200).render('index.html')
      
      } else {

        res.status(404).render('index.html')
      
      }

    })
  })
  
}