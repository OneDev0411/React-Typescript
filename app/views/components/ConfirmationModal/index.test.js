import React from 'react'

import { cleanup, fireEvent } from '@testing-library/react'

import ConfirmationModalContext from 'components/ConfirmationModal/context'
import renderWithConfirmationModal from 'utils/test-utils/render-with-confirmation-modal'

// Creating a test component with a button to open the modal.
function TestComponent(props) {
  const modal = React.useContext(ConfirmationModalContext)
  const [isPopUpOpen, setOpen] = React.useState(false)
  const overrideOptions = props.overrideOptions || {}

  function handleTestClick() {
    if (!isPopUpOpen) {
      modal.setConfirmationModal({
        message: 'Heads up!',
        description: 'You are missing something important. Are you sure?',
        onConfirm: () => {
          setOpen(!isPopUpOpen)
        },
        ...overrideOptions
      })
    } else {
      setOpen(!isPopUpOpen)
    }
  }

  return (
    <button data-test="test-button" type="button" onClick={handleTestClick}>
      Open Modal
    </button>
  )
}

afterEach(cleanup)

describe('ConfirmationModal component', () => {
  it('should render modal properly', async () => {
    const { container, getByTestId } = renderWithConfirmationModal(
      <TestComponent />
    )
    const btn = getByTestId('test-button')

    fireEvent.click(btn)
    expect(container).toMatchSnapshot()
  })

  it('should render anything by default', async () => {
    const { queryByTestId } = renderWithConfirmationModal(<TestComponent />)

    const cancelBtn = queryByTestId('confirmation-modal-cancel-button')
    const confirmBtn = queryByTestId('confirmation-modal-confirm-button')
    const title = queryByTestId('confirmation-modal-title')

    expect(cancelBtn).toBeNull()
    expect(confirmBtn).toBeNull()
    expect(title).toBeNull()
  })

  it('should use confirm options by consumer', async () => {
    const options = {
      confirmLabel: 'Boom!',
      onConfirm: jest.fn()
    }
    const { getByTestId } = renderWithConfirmationModal(
      <TestComponent overrideOptions={options} />
    )
    const btn = getByTestId('test-button')

    fireEvent.click(btn)

    const confirmBtn = getByTestId('confirmation-modal-confirm-button')

    fireEvent.click(confirmBtn)

    // Checking text of confirm button
    expect(confirmBtn.textContent).toBe('Boom!')
    // Checking it's called
    expect(options.onConfirm).toBeCalledTimes(1)
  })

  it('should use cancel options by consumer', async () => {
    const options = {
      cancelLabel: 'This should be canceled!',
      onCancel: jest.fn()
    }
    const { getByTestId, queryByTestId } = renderWithConfirmationModal(
      <TestComponent overrideOptions={options} />
    )
    const btn = getByTestId('test-button')

    fireEvent.click(btn)

    const cancelBtn = getByTestId('confirmation-modal-cancel-button')

    fireEvent.click(cancelBtn)

    const title = queryByTestId('confirmation-modal-title')

    // Checking text of cancel button
    expect(cancelBtn.textContent).toBe('This should be canceled!')
    // Checking it's called
    expect(options.onCancel).toBeCalledTimes(1)

    // it should be closed after pressing cancel
    expect(title).toBeNull()
  })
})
