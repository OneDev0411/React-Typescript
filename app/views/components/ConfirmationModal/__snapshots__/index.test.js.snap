// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal component should render modal properly 1`] = `
<div>
  <button
    data-test="test-button"
    type="button"
  >
    Open Modal
  </button>
  <div
    class="confirmation-title"
    data-test="confirmation-modal-title"
  >
    Heads up!
  </div>
  <div
    class="confirmation-descr"
    data-test="confirmation-modal-description"
  >
    You are missing something important. Are you sure?
  </div>
  <div
    class="styled__ActionBarContainer-sc-1w2c0cs-0 cSTkgI cta"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
      data-test="confirmation-modal-cancel-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Cancel
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-test="confirmation-modal-confirm-button"
      style="margin-left: 1em;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Confirm
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
